src/3d/framegraph/qgsambientocclusionrenderview.cpp

```diff
@@ -0,0 +1,183 @@
+#include <Qt3DRender/QLayerFilter>
+#include <Qt3DRender/QLayer>
+#include <Qt3DRender/QRenderTargetSelector>
+#include <Qt3DRender/QRenderTarget>
+#include <Qt3DRender/QTexture>
+#include <Qt3DRender/QClearBuffers>
+#include <Qt3DRender/qsubtreeenabler.h>
+#include <Qt3DRender/QCameraSelector>
+#include <Qt3DRender/QRenderStateSet>
+#include <Qt3DRender/QDepthTest>
+#include <Qt3DRender/QCullFace>
+
+
+QgsAmbientOcclusionRenderView::QgsAmbientOcclusionRenderView( const QString &viewName, Qt3DRender::QCamera *mainCamera, QSize mSize, Qt3DRender::QTexture2D *forwardDepthTexture, Qt3DCore::QEntity *rootSceneEntity )
+  : QgsAbstractRenderView( viewName )
+  , mMainCamera( mainCamera )
+{
+  mAOPassLayer = new Qt3DRender::QLayer;
+  mAOPassLayer->setRecursive( true );
+  mAOPassLayer->setObjectName( mViewName + "::Layer(AO)" );
+
+  mBlurPassLayer = new Qt3DRender::QLayer;
+  mBlurPassLayer->setRecursive( true );
+  mBlurPassLayer->setObjectName( mViewName + "::Layer(Blur)" );
+
+  // ambient occlusion rendering pass
+  buildRenderPass( mSize, forwardDepthTexture, rootSceneEntity );
+}
+
+void QgsAmbientOcclusionRenderView::updateWindowResize( int width, int height )
+{
+  mAOPassTexture->setSize( width, height );
+  mBlurPassTexture->setSize( width, height );
+}
+
+void QgsAmbientOcclusionRenderView::setEnabled( bool enable )
+{
+  QgsAbstractRenderView::setEnabled( enable );
+  if ( mAmbientOcclusionRenderEntity != nullptr )
+    mAmbientOcclusionRenderEntity->setEnabled( enable );
+  if ( mAmbientOcclusionBlurEntity != nullptr )
+    mAmbientOcclusionBlurEntity->setEnabled( enable );
+}
+
+Qt3DRender::QRenderTarget *QgsAmbientOcclusionRenderView::buildTextures( QSize mSize )
+{
+  // Create a texture to render into.
+  Qt3DRender::QRenderTargetOutput *colorTargetOutput = new Qt3DRender::QRenderTargetOutput;
+  colorTargetOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Color0 );
+
+  mBlurPassTexture = new Qt3DRender::QTexture2D( colorTargetOutput );
+  mBlurPassTexture->setSize( mSize.width(), mSize.height() );
+  mBlurPassTexture->setFormat( Qt3DRender::QAbstractTexture::R32F );
+  mBlurPassTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mBlurPassTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mBlurPassTexture->setObjectName( mViewName + "::ColorTarget" );
+  colorTargetOutput->setTexture( mBlurPassTexture );
+
+  Qt3DRender::QRenderTarget *renderTarget = new Qt3DRender::QRenderTarget;
+  renderTarget->addOutput( colorTargetOutput );
+
+  return renderTarget;
+}
+
+void QgsAmbientOcclusionRenderView::buildRenderPass( QSize mSize, Qt3DRender::QTexture2D *forwardDepthTexture, Qt3DCore::QEntity *rootSceneEntity )
+{
+  // AO pass
+  {
+    Qt3DRender::QCameraSelector *cameraSelector = new Qt3DRender::QCameraSelector( mRendererEnabler );
+    cameraSelector->setObjectName( mViewName + "::CameraSelector(AO)" );
+    cameraSelector->setCamera( mMainCamera );
```

Comment:
actually we should not need camera selector for both passes - they don't use the camera at all - let's remove it...

Reply:
ok strange but it works without QCameraSelector